ui/components/app/signature-request-siwe/signature-request-siwe.js

```diff
@@ -0,0 +1,180 @@
+import React, { useContext, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+import { EVENT } from '../../../../shared/constants/metametrics';
+import ActionableMessage from '../../ui/actionable-message';
+import Popover from '../../ui/popover';
+import Checkbox from '../../ui/check-box';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import { I18nContext } from '../../../contexts/i18n';
+import { PageContainerFooter } from '../../ui/page-container';
+import {
+  accountsWithSendEtherInfoSelector,
+  getSubjectMetadata,
+} from '../../../selectors';
+import { getAccountByAddress } from '../../../helpers/utils/util';
+import formatMessage from './format-message-params';
+import Header from './signature-request-siwe-header';
+import Message from './signature-request-siwe-message';
+
+export default function SignatureRequestSIWE({
+  txData,
+  cancelPersonalMessage,
+  signPersonalMessage,
+}) {
+  const allAccounts = useSelector(accountsWithSendEtherInfoSelector);
+  const subjectMetadata = useSelector(getSubjectMetadata);
+
+  const {
+    msgParams: {
+      from,
+      origin,
+      siwe: { parsedMessage },
+      version,
+    },
+    type,
+  } = txData;
+
+  const fromAccount = getAccountByAddress(allAccounts, from);
+  const targetSubjectMetadata = subjectMetadata[origin];
+
+  const t = useContext(I18nContext);
+  const trackEvent = useContext(MetaMetricsContext);
+
+  const isMatchingAddress = from === parsedMessage.address;
+
+  const checkSIWEDomain = () => {
+    let isSIWEDomainValid = false;
+
+    if (origin) {
+      const { host } = new URL(origin);
+      isSIWEDomainValid = parsedMessage.domain === host;
+    }
+    return isSIWEDomainValid;
+  };
+
+  const isSIWEDomainValid = checkSIWEDomain();
+
+  const [isShowingDomainWarning, setIsShowingDomainWarning] = useState(false);
+  const [agreeToDomainWarning, setAgreeToDomainWarning] = useState(false);
+  const onSign = (event) => {
+    signPersonalMessage(event);
+    trackEvent({
```

Comment:
Is the following thread proposing that we replace these trackEvents? 
https://consensys.slack.com/archives/C031PSFHQER/p1654811895007879?thread_ts=1654805714.261689&cid=C031PSFHQER

I think we can add a try/catch to handle `signPersonalMessage` and `cancelPersonalMessage`. These methods should be returning promises, and we'll want to await for the promises and confirm they have succeeded before tracking the events.

Reply:
Non-blocking. updating MetaMetrics in a future PR. For now, we updated the code to handle errors and include a @ todo comment
https://github.com/MetaMask/metamask-extension/pull/14438/commits/7b37892103297663236fb2aa869c5337525ad881